### ISSUE TITLE
Exception 'NSRangeException' when unregistering KVO

### ISSUE THREAD

**Issue by YanSte** [maxep/MXParallaxHeader#28]
Crash on   
- (void)willMoveToSuperview:(UIView *)newSuperview {
  [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext];
  }

this is my error : 
Terminating app due to uncaught exception 'NSRangeException', reason: 'Cannot remove an observer <(null) 0x0> for the key path "contentOffset" from <MXParallaxView 0x1569c8860> because it is not registered as an observer.'
**\* First throw call stack:
(0x180a42db0 0x1800a7f80 0x180a42cf8 0x1813590a8 0x181358b84 0x181358ac8 0x100ddaf80 0x185eb4014 0x185b994a0 0x185b9c924 0x181396cb0 0x1800c1ae8 0x1809209fc 0x1809f6bc0 0x180920c50 0x182208088 0x185c0a088 0x100281a58 0x1804be8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException

**Comment by sunnyyoung** [comment]
Same problem

**Comment by YanSte** [comment]
i try the try catch form but not work

**Comment by maxep** [comment]
Can you give a bit more context ? I cant reproduce on samples. Are you adding the parallax header to a UITableView ? I've made some changes recently that may have introduce this bug.

This does not work ? : 

``` objective-c
- (void)willMoveToSuperview:(UIView *)newSuperview {
    @try  {
        [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext];
    } @catch (NSException *exception) {}
}
```

Can you try this ? : 

``` objective-c
- (void)willMoveToSuperview:(UIView *)newSuperview {
    if (self.parent) {
        [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext];
    }
}

- (void)didMoveToSuperview {
    if (self.parent) {
        [self.superview addObserver:self.parent
                         forKeyPath:NSStringFromSelector(@selector(contentOffset))
                            options:NSKeyValueObservingOptionNew
                            context:kMXParallaxHeaderKVOContext];
    }
}
```

Thanks for the report.

**Comment by silvanosouzajunior** [comment]
same problem here

**Comment by leolelego** [comment]
@maxep the try/catch doesn't fix.

The second proposition give this : 

![2016-06-18 at 20 11 26](https://cloud.githubusercontent.com/assets/4069471/16172845/41fa6474-3591-11e6-9267-7ef7c9ae855e.png)

**Comment by wade0595** [comment]
same problem here

leave the view

libc++abi.dylib: terminate_handler unexpectedly threw an exception

**Comment by maxep** [comment]
Is v0.5.0 has this issue ? are you adding a parallax header to a table view?

**Comment by leolelego** [comment]
@maxep yes, this is a tableview from me, with static cell. And the app crash when I pop to the view before (UInavigationBar back button).

**Comment by silvanosouzajunior** [comment]
on the version v0.5.0 is working well

2016-06-20 14:54 GMT-03:00 Léo Derbois notifications@github.com:

**Comment by leolelego** [comment]
@maxep I just tried like this : `pod 'MXParallaxHeader','= 0.5.0'` And the crash was not in this release.

**Comment by developforapple** [comment]
```
- (void)willMoveToSuperview:(UIView *)newSuperview {
    [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext];
}

- (void)setTableHeaderViewFrame:(CGRect)frame {

    //Create a table header view that will raise KVO
    MXParallaxView *headerView = [[MXParallaxView alloc] initWithFrame:frame];
    headerView.parent = self;

    [headerView addSubview:self.contentView];
    [(UITableView *)self.scrollView setTableHeaderView:headerView];
    [headerView setNeedsLayout];
}
```

Conflict between the 2 methods above.  
self.contentView's class is MXParallaxView
 [headerView addSubview:self.contentView];  self.contentView's superView class also is MXParallaxView.

So [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext]; make crash.

My Code:

```
- (void)willMoveToSuperview:(UIView *)newSuperview {
    if ([self.superview isKindOfClass:[UIScrollView class]]) {
        [self.superview removeObserver:self.parent forKeyPath:NSStringFromSelector(@selector(contentOffset)) context:kMXParallaxHeaderKVOContext];
    }
}

- (void)didMoveToSuperview {
    if ([self.superview isKindOfClass:[UIScrollView class]]) {
        [self.superview addObserver:self.parent
                     forKeyPath:NSStringFromSelector(@selector(contentOffset))
                        options:NSKeyValueObservingOptionNew
                        context:kMXParallaxHeaderKVOContext];
    }
} 
```

**Comment by jj-12** [comment]
Good fix, but are there any plans of getting this into the next release of MXSegmentedPager?

**Comment by maxep** [comment]
Fix by #31
